server/src/main/scala/io/delta/sharing/server/DeltaSharingService.scala

```diff
@@ -237,6 +237,10 @@ class DeltaSharingService(serverConfig: ServerConfig) {
 
     val start = System.currentTimeMillis
     val tableConfig = sharedTableManager.getTable(share, schema, table)
+    if (queryTableRequest.version.isDefined && !tableConfig.cdfEnabled) {
+      throw new DeltaSharingIllegalArgumentException("reading table by version is not enabled " +
```

Comment:
Better to explain the root cause here: e.g. reading table by version is not supported because the table is not enabled with change data feed.

Reply:
+1

Reply:
updated.